packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -253,16 +253,23 @@ public class InAppPurchasePlugin: NSObject, FlutterPlugin, FIAInAppPurchaseAPI {
     let pendingTransactions = getPaymentQueueHandler().getUnfinishedTransactions()
 
     for transaction in pendingTransactions {
+      // finishTransaction() cannot be called on a Transaction with a current purchasing state
+      // https://developer.apple.com/documentation/storekit/skpaymentqueue/1506003-finishtransaction
+      guard transaction.transactionState != SKPaymentTransactionState.purchasing else {
+        return
```

Comment:
it should be `continue` rather than `return`?

Reply:
no - we want to check if the transaction is currently in the 'purchasing' state. If it is, stop further execution and return. transactions that are currently purchasing cannot be finished.

Reply:
the original code before your change was to continue if not meet condition?

Reply:
im so confused
before it was 

```
If (not purchasing) {
  finish()
}
```

now its 

```
guard (not purchasing) else {
   return 
 }

finish()
```

are these not equivalent or is my brain bugging out

Reply:
wait, isn't the whole chunk inside a for loop? 

Before: 

```
for ... {
  if condition {
    finish()
  }
}
```

After: 

```
for ... {
  guard condition else { 
    return
  }
  finish()
}
```

So in old code, if `condition` is not met, it will just go to the next loop; in new code, if `condition` is not met, it will return out of the function, rather than going to the next loop.

Reply:
ohhhhhhhHHHHHHH
I thought you were saying my logic was inversed, i was so confused